### PR TITLE
HDDS-4575: Refactor SCMHAManager and SCMRatisServer with RaftServer.Divison

### PR DESCRIPTION
**Pull request by GlenGeng**
## What changes were proposed in this pull request?

Simplify SCMHAManager and SCMRatisServer with RaftServer.Division

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4575

## How was this patch tested?

CI

### CODE REVIEW

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/ha/MockSCMHAManager.java [hunk 1]
```diff
@@ -204,23 +180,27 @@ private Message process(final SCMRatisRequest request)
...
     @Override
-    public RaftServer getServer() {
-      return null;
+    public void stop() {
     }
 
     @Override
-    public RaftGroupId getRaftGroupId() {
+    public RaftServer.Division getDivision() {
       return null;
```
Comment (timmylicheng) [inline hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/ha/MockSCMHAManager.java:188]: Do we allow Division to be null?

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMRatisServerImpl.java [hunk 2]
```diff
@@ -69,41 +65,49 @@
...
+
+    RaftServer server = RaftServer.newBuilder()
+        .setServerId(haGrpBuilder.getPeerId())
+        .setGroup(haGrpBuilder.getRaftGroup())
         .setProperties(serverProperties)
-        .setStateMachine(scmStateMachine)
+        .setStateMachine(new SCMStateMachine())
         .build();
+
+    this.division = server.getDivision(haGrpBuilder.getRaftGroupId());
```
Comment (timmylicheng) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMRatisServerImpl.java:80]: Any way to guarantee division is not null?

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/ha/MockSCMHAManager.java (hunk 1 above)
Comment (GlenGeng) [reply hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/ha/MockSCMHAManager.java:188]: No, we don't allow null Divison in real env.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMRatisServerImpl.java (hunk 2 above)
Comment (GlenGeng) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMRatisServerImpl.java:80]: You won't get a null, instead, the call will throw a `GroupMismatchException` in this situation.

**Comment by ChenSammi** [comment]
+1

**Pull request merged**